File: frontend/src/app/services/records.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class RecordsService {

  constructor(
    private _record:HttpClient
  ) { }

  // add record 
  public addRecord(record:any){
    return this._record.post(`${baseUrl}/user-activities/`, record);
  }

  // update record
  public updateRecord(record:any){
    return this._record.put(`${baseUrl}/user-activities/`, record);
  }

  // get all records
  public getAllRecords(){
    return this._record.get(`${baseUrl}/user-activities/`);
  }

  // get one record
  public getSingleRecord(aId:any){
    return this._record.get(`${baseUrl}/user-activities/${aId}`);
  }

  // delete record
  public deleteRecord(aId:any){
    return this._record.delete(`${baseUrl}/user-activities/${aId}`);
  }

  // get records of quiz
  public getRecordsOfQuiz(qId:any){
    return this._record.get(`${baseUrl}/user-activities/quiz/${qId}`);
  }

  // get records of user
  public getRecordsOfUser(username:any){
    return this._record.get(`${baseUrl}/user-activities/user/${username}`);
  }
}

File: backend/src/main/java/com/classstudies/classquiz/service/RecordsService.java
package com.classstudies.classquiz.service;

import java.util.Set;

import com.classstudies.classquiz.model.User;
import com.classstudies.classquiz.model.exam.AttemptedQuizRecords;
import com.classstudies.classquiz.model.exam.Quiz;

public interface RecordsService {
	
	public AttemptedQuizRecords addRecord(AttemptedQuizRecords record);
	
	public AttemptedQuizRecords updateRecord(AttemptedQuizRecords record);
	
	public Set<AttemptedQuizRecords> getRecords();
	
	public AttemptedQuizRecords getRecord(Long recordId);
	
	public void deleteRecord(Long recordId);
	
	public Set<AttemptedQuizRecords> getRecordsOfUser(String username);
	
	public Set<AttemptedQuizRecords> getRecordsOfQuiz(Quiz quiz);

}
